### PR TITLE
CI: Test integration_tests with Python 3.9, 3.10 and 3.11

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lcompilers/lpython/issues/1827#issuecomment-1692612951

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
It seems `linear_regression`, `correlation`, `covariance` were added to the statistics module in `Python 3.10` (https://docs.python.org/3/library/statistics.html#statistics.correlation). I temporarily removed them in [tmp](https://github.com/lcompilers/lpython/pull/2298/commits/107355be4db1bf56321edeb5d7d6f8826f13d45b).

**Comment by certik** [comment]
I have an idea. Why not to test `integration_tests/test_statistics.py` only with Python >= 3.10. That way we'll have a mechanism to test features in newer CPython versions. In LPython we can support the latest features, or we can even allow this with compiler options (but I would not bother for now, we can do that later).

**Comment by Shaikh-Ubaid** [comment]
We currently test both `cpython` and `c_py` (as the test depends on CPython for execution). We will eventually support other backends like `llvm_py` and probably `c_sym`, `llvm_sym`.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -623,7 +635,7 @@ RUN(NAME bindpy_01           LABELS cpython c_py ENABLE_CPYTHON NOFAST COPY_TO_B
 RUN(NAME bindpy_02           LABELS cpython c_py LINK_NUMPY COPY_TO_BIN bindpy_02_module.py)
 RUN(NAME bindpy_03           LABELS cpython c_py LINK_NUMPY NOFAST COPY_TO_BIN bindpy_03_module.py)
 RUN(NAME bindpy_04           LABELS cpython c_py LINK_NUMPY NOFAST COPY_TO_BIN bindpy_04_module.py)
-RUN(NAME bindpy_05           LABELS llvm_py c_py ENABLE_CPYTHON COPY_TO_BIN bindpy_05_module.py)
+RUN(NAME bindpy_05           LABELS llvm_py c_py ENABLE_CPYTHON COPY_TO_BIN bindpy_05_module.py REQ_PY_VER 3.10)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:638]: This test seems to fail with `Python 3.9`. For the moment, I marked it as `REQ_PY_VER 3.10`. I will tackle the failure with `Python 3.9` in a future PR.

**Comment by Shaikh-Ubaid** [comment]
From https://github.com/lcompilers/lpython/issues/1827#issuecomment-1692612951,

>Ok, then let's test Python 3.9, 3.10 and 3.11 explicitly, all three

We now explicitly test all the three `3.9`, `3.10` and `3.11` Python versions with `integration_tests`.

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -334,6 +334,18 @@ macro(RUN)
         set(RUN_EXTRA_ARGS ${RUN_EXTRA_ARGS} -I${CMAKE_CURRENT_SOURCE_DIR}/${RUN_IMPORT_PATH})
     endif()
 
+    if (RUN_REQ_PY_VER)
+        string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)" "\\1" PY_MAJOR_VERSION "${Python_VERSION}")
+        string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)" "\\2" PY_MINOR_VERSION "${Python_VERSION}")
+        string(REGEX REPLACE "([0-9]+)\\.([0-9]+)" "\\1" REQ_PY_MAJOR_VERSION "${RUN_REQ_PY_VER}")
+        string(REGEX REPLACE "([0-9]+)\\.([0-9]+)" "\\2" REQ_PY_MINOR_VERSION "${RUN_REQ_PY_VER}")
+
+        if (PY_MINOR_VERSION LESS REQ_PY_MINOR_VERSION)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:343]: This compares only the Python minor versions for now. We might need to consider Python major versions when Python `4.0.0` is released, but I guess there might probably be some time for it.

**Comment by Shaikh-Ubaid** [comment]
The CI is green. This is ready!

**Review by certik — APPROVED** [review]
I think this is good, thank you for fixing it!

**Pull request merged**